diff --git a/tensorflow/workspace0.bzl b/tensorflow/workspace0.bzl
index d8b53e1ef8b..a8c00f1598e 100644
--- a/tensorflow/workspace0.bzl
+++ b/tensorflow/workspace0.bzl
@@ -133,7 +133,7 @@ def workspace():
     _tf_bind()
 
     grpc_extra_deps()
-    rules_foreign_cc_dependencies()
+    rules_foreign_cc_dependencies(cmake_version="3.29.5")
     config_googleapis()
 
 # Alias so it can be loaded without assigning to a different symbol to prevent
diff --git a/tensorflow/workspace3.bzl b/tensorflow/workspace3.bzl
index af1613994a7..890ebab2c92 100644
--- a/tensorflow/workspace3.bzl
+++ b/tensorflow/workspace3.bzl
@@ -62,6 +62,13 @@ def workspace():
     # but provides a script for setting up build rules via overlays.
     llvm("llvm-raw")
 
+    http_archive(
+        name = "rules_foreign_cc",
+        sha256 = "4b33d62cf109bcccf286b30ed7121129cc34cf4f4ed9d8a11f38d9108f40ba74",
+        strip_prefix = "rules_foreign_cc-0.11.1",
+        url = "https://github.com/bazelbuild/rules_foreign_cc/releases/download/0.11.1/rules_foreign_cc-0.11.1.tar.gz",
+    )
+
 # Alias so it can be loaded without assigning to a different symbol to prevent
 # shadowing previous loads and trigger a buildifier warning.
 tf_workspace3 = workspace
